plugins/crd/crd_queue.go

```diff
@@ -49,39 +53,53 @@ func networkserviceWork(plugin *Plugin) {
 		// We define a function here to process a queue item, so that we can
 		// use 'defer' to make sure the message is marked as Done on the queue
 		func(key string) {
-			var obj interface{}
-			var err error
-
 			defer queueNS.Done(key)
 
 			// Attempt to split the 'key' into namespace and object name
 			namespace, name, err := cache.SplitMetaNamespaceKey(strKey)
 
 			if err != nil {
+				// This is a soft-error, retry up to QUEUE_RETRY_COUNT times
 				plugin.Log.Errorf("Error splitting meta namespace key into parts: %s", err.Error())
-				runtime.HandleError(fmt.Errorf("Error splitting meta namespace key into parts: %s", err.Error()))
-				// This is a soft-error, we merely want to forget it and mark it done on the queue
-				queueNS.Forget(key)
+				if err != nil {
+					if queueNS.NumRequeues(key) < QUEUE_RETRY_COUNT {
+						plugin.Log.Errorf("Error processing item with key %s, error %v", key, err)
```

Comment:
Is there a reason we need to retry when `SplitMetaNamespaceKey` method returns an error, because all this method is doing is a key split and everytime we retry `key` remains the same, it will keep failing. Unless I am missing something here.

Reply:
This is a valid point, I'll re-spin removing the retry here.